### PR TITLE
Corrected/amended diagrams, explanations and code

### PR DESCRIPTION
**Pull request by michhar**
Several fixes:

- Using corrected BIO schema on data now
- Better explanations of CRF and Viterbi with new diagram of a more realistic use of Viterbi on a linear-CRF
- Fixed code to look up indices for words based on the entire vocabulary (recipe)
- Fixed CRF transition matrix diagram to reflect amended schema
- Showing output now

Comments welcome.  Needs review.

### CODE REVIEW

File: Named_Entity_Recognition/NamedEntityRecognition.ipynb [hunk 1]
```diff
@@ -800,35 +997,44 @@
...
+    }
+   ],
    "source": [
     "with torch.no_grad():\n",
     "    precheck_sent = prepare_sequence(test_data[0][0], word_to_ix)\n",
-    "    print(model(precheck_sent))\n",
-    "    print(test_data[0][1])\n",
-    "    print(test_data[0][0])"
+    "    pred =  model(precheck_sent)[1]\n",
+    "    print('Prediciton:   ', [ix_to_tag[idx] for idx in pred])\n",
```
Comment (kendallc) [inline Named_Entity_Recognition/NamedEntityRecognition.ipynb]: Typo: Prediciton

**Comment by kendallc** [comment]
Found a small typo but looks good overall!

File: Named_Entity_Recognition/NamedEntityRecognition.ipynb [hunk 2]
```diff
@@ -152,7 +270,7 @@
    "source": [
     "#### Named Entity Recognition Task\n",
     "\n",
-    "For the task of **Named Entity Recognition (NER)** is helpful to have context from past as well as the future, or left and right contexts.  This can be addressed with a BLSTM which is two LSTMs, one processing information in a forward fashion as we have already seen and another LSTM that processes the sequences in a reverse fashion giving the future context.  _That second LSTM is just reading the sentence in reverse._  \n",
+    "For the task of **Named Entity Recognition (NER)** is helpful to have context from past as well as the future, or left and right contexts.  This can be addressed with a Bi-LSTM which is two LSTMs, one processing information in a forward fashion as we have already seen and another LSTM that processes the sequences in a reverse fashion giving the future context.  _That second LSTM is just reading the sentence in reverse._  \n",
```
Comment (dmclark53) [inline Named_Entity_Recognition/NamedEntityRecognition.ipynb]: Missing 'it' before 'is' and after '(NER)'.

File: Named_Entity_Recognition/NamedEntityRecognition.ipynb [hunk 3]
```diff
@@ -167,9 +285,9 @@
    "source": [
     "### Conditional Random Field\n",
     "\n",
-    "We don't have to stop at the output vector from the BLSTM!  We're not to our tag for the entity, yet.  We need to understand costs of moving from one tag to the next (or staying put on a tag, even).\n",
+    "We don't have to stop at the output vector from the Bi-LSTM!  We're not to our tag for the entity, yet.  We need to understand costs of moving from one tag to the next (or staying put on a tag, even).\n",
```
Comment (dmclark53) [inline Named_Entity_Recognition/NamedEntityRecognition.ipynb]: In the second sentence 'to' should be 'at'.

**Comment by michhar** [comment]
@dmclark53 @kendallc Thanks.  All fixed.

**Pull request merged**